modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/GlassViewEventHandler.java

```diff
@@ -669,9 +669,12 @@ protected void onChanged(ListChangeListener.Change<InputMethodTextRun> c) {
     @Override
     public double[] getInputMethodCandidatePos(int offset) {
         Point2D p2d = scene.inputMethodRequests.getTextLocation(offset);
+        final Screen s = Screen.getMainScreen();
```

Comment:
This won't work for multiple screens. You'll need something like `scene.getPlatformView().getWindow().getScreen()` (with the appropriate null checks).